sqlitecluster/SQLiteNode.cpp

```diff
@@ -2580,6 +2541,77 @@ void SQLiteNode::_recvSynchronize(Peer* peer, const SData& message) {
         throw "commits remaining at end";
 }
 
+bool SQLiteNode::peerClosenessSort(const Peer* lhs, const Peer* rhs)
+{
+    // Null pointers are never fast/close.
+    if (!lhs) {
+        return false;
+    }
+    if (!rhs) {
+        return true;
+    }
+
+    // If two peers have the same latency, then the one with the higher CommitCount is closest. This case is most
+    // likely in the event that we have no latency data for either peer.
+    if (lhs->latency == rhs->latency) {
+        return SToInt((*lhs)["CommitCount"]) > SToInt((*rhs)["CommitCount"]);
+    }
+
+    // If only one peer has 0 latency, the other is faster/closer.
+    if (!lhs->latency) {
+        return false;
+    }
+    if (!rhs->latency) {
+        return true;
+    }
+
+    // Both peers have latency data, and neither are 0.
+    return lhs->latency < rhs->latency;
+}
+
+SQLiteNode::Peer* SQLiteNode::updateSyncPeer()
+{
+    // Sets are stored in sorted order. This set uses peerClosenessSort as its comparator function.
+    set<Peer*, decltype(&peerClosenessSort)> latencies {peerClosenessSort};
```

Comment:
I don't love this either, but it's sort of a C++11-y way to do [this](https://github.com/Expensify/Bedrock/blob/master/libstuff/libstuff.h#L103), which I also don't like. The way we do it there requires a whole new class, whereas this way lets you jsut use a normal function, but the instantiation is uglier.

I might just re-write this without sorted set and avoid the whole thing.